daisy/error.go

```diff
@@ -77,8 +77,13 @@ func Errf(format string, a ...interface{}) DError {
 }
 
 // wrapErrf returns a DError by keeping errors type and replacing original error message.
-func wrapErrf(e DError, format string, a ...interface{}) DError {
-	return &dErrImpl{errs: []error{fmt.Errorf(format, a...)}, errsType: e.errorsType(), anonymizedErrs: []string{format}}
+func wrapErrf(e DError, formatPrefix string, a ...interface{}) DError {
+	f := formatPrefix + strings.Join(e.AnonymizedErrs(), "; ")
+	return &dErrImpl{
+		errs:           []error{fmt.Errorf(fmt.Sprintf(formatPrefix, a...) + e.Error())},
```

Comment:
What if formatPrefix doesn't include a space? Will the resulting error have e.Error() stuck at the end without any space?

Reply:
You mean formatPrefix doesn't include a tailing space? Then it's an ugly format. Functionality won't be impacted.
There isn't necessarily a space there. For example, maybe people want to format it as "!!!SEERIOUS_ERROR>>>%v"

Reply:
I mean, in: 
fmt.Sprintf(formatPrefix, a...) + e.Error()

If format error is "Serious error in value `%v`" to which you concatenate e.Error(), formatting might get ugly, like: "Serious error in value `hostname`InvalidResourceException"

Reply:
If they format it like that, then yes. Would that be a problem? People can ruin any format...

Reply:
Whoever is calling wrapErrf might not be aware how error will be formatted. It's safe to assume formatPrefix will be used to format a..interface{} values, but it's not clear error will be appended as well. That's why I think we should add a space or any other separator between the formatted formatPrefix and e.Error()